### ISSUE TITLE
Update the Community Users Screen Design and Functionality

### ISSUE THREAD

**Issue by palisadoes** [PalisadoesFoundation/talawa-admin#1452]
### NOTE

1. This issue will require close coordination with persons assigned issues in the reference section.
2. You may find it easier to split responsibilities in a different way than proposed here to be more efficient is creating the goal

**Is your feature request related to a problem? Please describe.**

The current _COMMUNITY_ `Users` screen needs to be updated. It does not reflect the soon to be implemented functionality of Admins being assigned per organization.

![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/16875803/b799fb5f-edf7-4ba7-b307-0071dff4acd5)

**Describe the solution you'd like**
1. Use the Figma file in our Design system document for guidance
    1. https://docs.talawa.io/docs/design/ux/ux-design-system
1. Use the People screen components from the first PR of this issue:
    1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1451
    3. The resulting screen would look like this
![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/16875803/03b27afc-5cdb-4491-ab0d-67e6e0a961b7)
1. Work with the person assigned the same issue to create an Organizations tab on the User Detail page. A small follow up PR with a skeleton design should be sufficient.
   1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1451
1. Clicking on a User will go to the first profile page
1. Clicking on an `Organization` tab will go to a page listing the person's `Joined` and `Blocked` organizations designed like this with the following modifications:
    1. The `Created Organizations` functionality is not required at this time
    2. The sort button must be to the right of the search bar like in most of our other screens
    4. You will need to propose the best way to order `Joined` and `Blocked` organizations on this screen.
    5. You will need to only show the organizations to which the user belongs or from which they have been blocked.
    6. The `Manage` button text must be replaced by `Edit Role` (See Blue Box below)
    ![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/16875803/87caf52f-930e-4374-b8db-267390ece5bf)
1. Clicking on the newly named `Edit Role` button will present the existing `View Organization` dialog that is currently visible on the `Community Users` screen.
    1. In this dialog, as before:
        1. the user's role can be changed 
        1. the user can be removed from the organization
    1. There will be no need to search for organizations as the User will be editing access for a single organization.
    ![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/16875803/8f059332-1152-4a34-a6c5-348a4d28cdda)

**Describe alternatives you've considered**

- N/A

**Approach to be followed (optional)**

Work with the following contributors for guidance:

1. @aashimawadhwa
3. @EshaanAgg 
4. @beingnoble03 
5. @rishav-jha-mech

Also:

1. We will need to add your Figma page designs to our official Figma document
2. Please plan for this

**Additional context**

Related Issues and PRs:

1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1451
7. https://github.com/PalisadoesFoundation/talawa-admin/issues/1452
8. https://github.com/PalisadoesFoundation/talawa-admin/issues/1445
1. https://github.com/PalisadoesFoundation/talawa-api/issues/1862

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship
- https://github.com/PalisadoesFoundation/talawa/issues/359

**Comment by Sejal1411** [comment]
would like to work on this issue. Pls assign!

**Comment by SanjeevKrSingh** [comment]
Hey @palisadoes  I like to work on this issue could you please assign this issue to me

**Comment by Cioppolo14** [comment]
@SanjeevKrSingh If the issue has already been assigned, please don't ask to be assigned. We want everyone to get a chance.

**Comment by palisadoes** [comment]
@Sejal1411 

1. We will need to add your Figma page designs to our official Figma document
2. Please plan for this

**Comment by Sejal1411** [comment]
@palisadoes  do i need to design from scratch or make modifications in the official copy and submit? Also i have some queries...

1. The User's screen will resemble the one in Figma file?
2. No limitation to  a person joining the organisation?

**Comment by palisadoes** [comment]
1. The Figma file in the official documentation should be your guide. Make a copy to create your design.
2. There isn't a join organization screen to be designed. You will need to only show the organizations to which the user belongs.

**Comment by Sejal1411** [comment]
@palisadoes I have created some designs...Need your feedback on them..How should i proceed?

**Comment by palisadoes** [comment]
Post the images here so we can all see

**Comment by Sejal1411** [comment]
![Screenshot (278)](https://github.com/PalisadoesFoundation/talawa-admin/assets/106868816/ecb1fd2c-9f7a-4e9b-8ff4-82f61d5b36eb)
![Screenshot (279)](https://github.com/PalisadoesFoundation/talawa-admin/assets/106868816/a03c6743-4582-49e2-ad9b-febc83ce5dd1)
![Screenshot (280)](https://github.com/PalisadoesFoundation/talawa-admin/assets/106868816/5fbf7784-e942-42f0-88c4-18bd4b590ead)


Should I proceed with this design?

**Comment by palisadoes** [comment]
1. In hindsight, only super admins should be able to see all the organizations a person is a member of. 
2. Therefore a click through of the screen in the image 1 should show the screen of image 2, with all the organizations the person is a part of. As it is  a super admin screen, the left drawer of image 2 should be the same as image 1. Keep the Overview tab. We can fill in the Overview tab later with the design from user profile PR.
3. In image 2, sort alphabetically by default. Instead of "sort by", make the drop down filter "all", "joined", "blocked" and "available" organizations.
4. I'd like you to figure out a way to use the screen of image 2 as a starting point to:
    1. remove people from, or add people to, available organizations. 
    1. change the roles of people in the organization selected. 
    2. The screen that does this now combines two things in one in a way that is not intuitive. For example:
        1. there is no "save" or "submit"  button.
        2. Knowing the creation details of the organization isn't necessary for this pop up
    ![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/16875803/8f059332-1152-4a34-a6c5-348a4d28cdda)

This is a good first step. Thanks for the effort.

When this is done, we can open a second PR for this issue where clicking on the people button, on the organization specific page, will give you the profile overview screen with a tab for organizations. However in this case as it is an organization specific screen,  you should just show a single organization, namely itself (the Palisadoes organization). As it's only one organization there will be no need for filters and search. Just like you have, the admin should be able to promote or demote someone, but NOT to Super Admin, as this is an Admin only screen.

**Comment by Sejal1411** [comment]
okay @palisadoes, Noted! Also, I have some queries:-

1.  But if a the person is an USER/ADMIN, then he will be able to see "Overview" section only.
2.  How do you categorise "Available" organisations? Those orgs which do not have any members, or Admins or contain only users no Admin?

**Comment by palisadoes** [comment]
1. "Available" would be organizations that the user's profile isn't currently a member of.
2. Talawa Admin doesn't work the way as intended. 
    1. Only Super Admins should be able to see user information button in image 1. They will be able to see ALL organizations in the community.
    2. Admins should NOT see the user image button in image 1. Admins are supposed to only see information related to the organizations for which they have responsibility, not ALL.  
1. This means to eventually implement this logic is being addressed with this issue:
    1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1445

Focus on this for your first PR. The screens for the specific organization can come later

**Comment by Sejal1411** [comment]
Hey @palisadoes, I have created the following designs :-

1. As you mentioned, in Image 1, the SUPERADMIN will be able to view all the organisations a person is a member of.

![Screenshot 2024-01-27 194813](https://github.com/PalisadoesFoundation/talawa-admin/assets/106868816/c25671fc-14d3-44e2-8cde-da642bf84948)

2. Image 2 combines 2 functionalities:- changing roles and removing people from available organisations.
3. In Image 3 I have added "Add User" functionality to available organisation.

![Screenshot 2024-01-27 194829](https://github.com/PalisadoesFoundation/talawa-admin/assets/106868816/3be083bc-04fa-4650-8bcf-af081dcd31b4)

Please review these changes and suggest any changes if required!

**Comment by Sejal1411** [comment]
Hey @palisadoes, I have created the following designs :-

1. As you mentioned, in Image 1, the SUPERADMIN will be able to view all the organisations a person is a member of.

![Screenshot 2024-01-27 200502](https://github.com/PalisadoesFoundation/talawa-admin/assets/106868816/47aece66-b567-481b-8422-02cba4a369df)

2. Image 2 combines 2 functionalities:- changing roles and removing people from available organisations.

![Screenshot 2024-01-27 200822](https://github.com/PalisadoesFoundation/talawa-admin/assets/106868816/41516413-66bf-4d64-8ffd-f4673f52b23a)

3. In Image 3 I have added "Add User" functionality to available organisation.

![Screenshot 2024-01-27 200902](https://github.com/PalisadoesFoundation/talawa-admin/assets/106868816/56afcefb-b0d8-46a9-8723-3bca0fed193c)


Please review these changes and suggest any changes if required!

**Comment by palisadoes** [comment]
1. As mentioned before you need to focus on the Super Admin focused area where the left drawer looks like this where the `Users` button is highlighted not `People`:
    ![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/16875803/e69906dd-d696-4586-ac23-c7a9ecd5f940)
1. As only one user is previously highlighted, then only that user should be visible. Here you have 3. As it is a list of organizations for which roles need to be updated, it would be better just to have a list, versus the cards with an edit role link where the pop up becomes visible
![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/16875803/97b4edf1-e5c4-40d6-809f-d70cf93fc803)

**Comment by palisadoes** [comment]
Using the same cards is going to cause confusion as it looks too much like the select organization screen for the super admin

**Comment by Sejal1411** [comment]
Hey @palisadoes I have modified the designs. Please have a look and suggest changes if required!

1.In Image1,  when the SUPERADMIN clicks the Users section he is taken to "Joined Orgs" by the User.

![Screenshot 2024-01-30 194544](https://github.com/PalisadoesFoundation/talawa-admin/assets/106868816/b49e8511-7c5e-4308-a037-901d0f1b8c49)

2. Next, In Image 2 if he clicks "Available Org" he is redirected to this page where he can see all the Organisations User is not a part of. When he clicks "View" button...The left section appears where Users can be added by filling the details.
![Screenshot 2024-01-30 194614](https://github.com/PalisadoesFoundation/talawa-admin/assets/106868816/c2de7477-3b03-4744-8b08-000bcdff69c9)

3. In Image3, Other Users joined to this org can be searched, hence role modification and deletion of these users can be done.

![Screenshot 2024-01-30 194634](https://github.com/PalisadoesFoundation/talawa-admin/assets/106868816/38114f85-6e27-4bbb-b7af-e89c0baf6307)

**Comment by palisadoes** [comment]
@rishav-jha-mech @aashimawadhwa We are going to need your assistance in creating the design screens. It will make the work of @Sejal1411 easier

**Comment by Sejal1411** [comment]
@aashimawadhwa @EshaanAgg could you please review my design with the above suggested modifications...and how should i implement "Edit" button further?

[https://www.figma.com/file/ExFxla0Ew7VOzTxboJb0sA/Talawa-Design-System-(Official)-(Copy)?type=design&node-id=307-212&mode=design&t=26RhT7fmlKPbMkhv-0](url)

**Comment by Sejal1411** [comment]
@aashimawadhwa @EshaanAgg could you please review my design with the above suggested modifications...and how should i implement "Edit" button further?

https://www.figma.com/file/ExFxla0Ew7VOzTxboJb0sA/Talawa-Design-System-(Official)-(Copy)?type=design&node-id=307-212&mode=design&t=26RhT7fmlKPbMkhv-0

**Comment by palisadoes** [comment]
Thanks. The text should be edit. Otherwise this is OK.

![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/16875803/0bbcaebb-6b80-47c6-9059-cba007d9fee4)

How will you handle the editing options I mentioned previously in your design?

**Comment by Sejal1411** [comment]
Yes. Will this be done through popup? Or some other way!

**Comment by palisadoes** [comment]
Use your initiative and provide us with a solution.

**Comment by palisadoes** [comment]
1. Explain what the red icon does.
2. Why are the member and non-member rows different? This is not desirable. There needs to be a single screen for updating the role and status of a user.

**Comment by palisadoes** [comment]
@rishav-jha-mech @aashimawadhwa We need your assistance here

**Comment by Sejal1411** [comment]
1. It deletes the user.
2. It adds the person if he is a non-member. If he is a member the SuperAdmin can delete the same. (Sorry for the choice of the color). Sir, Also I am a bit confused regarding the execution flow. Please guide me !

**Comment by palisadoes** [comment]
Execution flow:
- https://github.com/PalisadoesFoundation/talawa-admin/issues/1452#issuecomment-1917623695

**Comment by Sejal1411** [comment]
Sir, correct me if I am wrong!

1. For different sorting option, the same tabular format would be maintained?

 For eg:- "Joined Org" option will have :

-  Organization Name-> XYZ
- User Name-> the name of the User, Superadmin clicked on the "Users" section?
- Status-> Member (as he is joined to the org)
- Role-> (the role alloted to him)
- Edit-> Superadmin can either delete him or change the role.

**Comment by palisadoes** [comment]
@rishav-jha-mech @aashimawadhwa We need your assistance here

**Comment by aashimawadhwa** [comment]
> Sir, correct me if I am wrong!
> 
> 1. For different sorting option, the same tabular format would be maintained?
> 
> For eg:- "Joined Org" option will have :
> 
> * Organization Name-> XYZ
> * User Name-> the name of the User, Superadmin clicked on the "Users" section?
> * Status-> Member (as he is joined to the org)
> * Role-> (the role alloted to him)
> * Edit-> Superadmin can either delete him or change the role.

Yes this flow looks fine to me. @Sejal1411

**Comment by Sejal1411** [comment]
Also, clicking on the "Edit" button -> should it have a separate screen or like this?

![Screenshot 2024-02-03 103553](https://github.com/PalisadoesFoundation/talawa-admin/assets/106868816/bfebeaed-bd0d-4794-ada8-3257d5162ff9)

**Comment by rishav-jha-mech** [comment]
@Sejal1411 @palisadoes 

I will be posting the new design soon....

**Comment by Sejal1411** [comment]
@rishav-jha-mech Do I have to make any modifications in the above design?

**Comment by Cioppolo14** [comment]
Unassigning due to no activity or open PR.

**Comment by NamitBhutani** [comment]
I would like to work on this feature request!

**Comment by Cioppolo14** [comment]
@NamitBhutani Our policy is to assign no more than two issues to each contributor across all repositories. This way everyone gets a chance to participate in the projects. We sometimes give exceptions for more urgent cases and sometimes we lose track, but the policy stands. You have reached your limit, please wait until your existing issues are closed before requesting more issues. You could unassign yourself from one of the other issues too.

**Comment by kanhaiya04** [comment]
@Cioppolo14 I want to work on this.

**Comment by kanhaiya04** [comment]
@rishav-jha-mech Is the design done, so that I can start the implementation.

**Comment by palisadoes** [comment]
> @rishav-jha-mech Is the design done, so that I can start the implementation.

1. Use the design elements of this PR, they are mostly the same.
    1. https://github.com/PalisadoesFoundation/talawa-admin/pull/1568
1. The Figma design linked in this issues original comment should also be used.

**Comment by kanhaiya04** [comment]
> The nature of the relationship of Users and Organizations was implemented incorrectly. The above is a stop gap measure. We'll need to revisit this when the relationship is fixed

@palisadoes I don't think the backend is implemented properly, as there is no `userType` per person per organization which is important for this new users screen design.

**Comment by palisadoes** [comment]
It is on the `develop-fixUserType` branch in the API

There is an issue to make Admin compatible which has already had a few PRs.

You can coordinate with the person assigned that issue

**Comment by palisadoes** [comment]
This is one of the last open issues before we merge back into `develop`

Are we on track for a PR this weekend?

**Comment by AVtheking** [comment]
@kanhaiya04 if you have any problem in implementing the api's then contact me ,we need to resolves this bug quickly

**Comment by palisadoes** [comment]
Only 2 issues to go before we are ready to merge back into develop. This is one of them.

**Comment by kanhaiya04** [comment]
@palisadoes, I've addressed the issues on the `Users screen` to align with the fixed `API` in this [PR](https://github.com/PalisadoesFoundation/talawa-admin/pull/1716), except for the role field. Feel free to merge it into the `develop` branch. Once the redesign is complete, I'll create another PR in develop branch.

**Comment by palisadoes** [comment]
OK. So we only have one PR left to go.

**Comment by palisadoes** [comment]
Should this be closed? `develop-userTypeFix` is now merged into `develop`
